builds/jenkins-build-pack-api/Jenkinsfile

```diff
@@ -347,6 +348,9 @@ node() {
               lambdaARN = getLambdaARN(stackName, credsId);
               events.sendCompletedEvent('CREATE_ASSET', null, utilModule.generateAssetMap("aws", lambdaARN.arn, "lambda", config), environment_logical_id);
 
+              def iamRoleArn = utilModule.getRoleDetails(lambdaARN.arn, config['region'], credsId)
+              events.sendCompletedEvent('CREATE_ASSET', null, utilModule.generateAssetMap("aws", iamRoleArn, "iam_role", config), environment_logical_id);
```

Comment:
It looks like that we are not creating the role as part of Jazz deployment workflow, why would we need to persist it as an asset?

Reply:
Correction - We do create it as part of serverlessDeploy before this step but shouldn't we send this event after reading the roleArn from deployment output instead of this additional call to aws lambda service?

Reply:
Also, we make call to CloudFormation (as part of `getLambdaARN` function) previously, can we leverage that and pull other resource information during the same call?

Reply:
We will not get all the stack information from the "cloudformation describe-stacks" . It does not contain the role arn. So we might need the other function to get the role details.

Reply:
That's weird.. Aren't we creating these roles as part of the serverless.yml deployment (if os, it will be a part of the  same CloudFormation template)..

Reply:
Please see the logs in the console http://52.90.194.98:8081/job/build-pack-api/40/console

Reply:
@SiniWilson  Should we use  [describe-stack-resources](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/describe-stack-resources.html)  instead of `describe-stacks`?